xclim/indices/_streamflow_flow_indices.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import annotations
+
+from xclim.core.units import declare_units
+
+
+@declare_units(q="[discharge]")
+def flow_index(q: xr.DataArray, p: float = 0.95) -> xr.DataArray:
+    """
+    Calculate the Qp  (pth percentile of daily streamflow) normalized by the mean flow.
+
+    Reference:
+    1. Addor, Nans & Nearing, Grey & Prieto, Cristina & Newman, A. & Le Vine, Nataliya & Clark, Martyn. (2018). A Ranking of Hydrological Signatures Based on Their Predictability in Space. Water Resources Research. 10.1029/2018WR022606.
+    2. Clausen, B., & Biggs, B. J. F. (2000). Flow variables for ecological studies in temperate streams: Groupings based on covariance. Journal of Hydrology, 237(3–4), 184–197. https://doi.org/10.1016/S0022-1694(00)00306-1
```

Comment:
For references, we try to make use of `sphinxcontrib-bibtext`. This involves adding BibTeX entries to https://github.com/Ouranosinc/xclim/blob/main/docs/references.bib and citing them using the `cite:cts:`authornameyear` operator.